File: eau04.py
"""
Créez un programme qui affiche le N-ème élément de la célèbre suite de Fibonacci. (0, 1, 1, 2) étant le début de la suite et le premier élément étant à l’index 0.


Exemples d’utilisation :
$> python exo.py 3
2
$>

Afficher -1 si le paramètre est négatif ou mauvais.

"""

"""

def suite_fibo(user):
    a, b, c, = 0, 1 ,1
    while c != user :
        print(b, end=" ")
        a,b,c= b, a+b, c+1
    try: 
        valeur_str = int(valeur_str)
    except:
        print("ereur")
        exit(1)
        

user= int(input("entrer un argument"))


"""

#n=input("Entré un chifre pour calculer Le N'eme élément de la suite de fibonancy ")

def fib(n):
    if n <=2:
        result=1
    elif n <=0 :
        print("ereur -1")
        exit(1)
    elif n =="":
        print("Erreur -1")
        exit
    else:
        result= fib(n-1) + fib(n-2)
    return result

n=int(input("rentré un chiffre pour calculer la n'emme élément de la suite de fibonancy "))

present=print(fib(n))

File: eau04.js
/*
Créez un programme qui affiche le N-ème élément de la célèbre suite de Fibonacci. (0, 1, 1, 2) étant le début de la suite et le premier élément étant à l’index 0.
Exemples d’utilisation :
$> python exo.py 3
2
$>

Afficher -1 si le paramètre est négatif ou mauvais.
*/


//Parsing
function getArguments () {
    const arguments = process.argv.slice(2);
    return arguments;
}

//Gestion d'erreur
function isValidArguments (arguments) { 
    const num = Number(arguments[0])
    if (isNaN(num)){
    console.error("l'argument doit être un nombre.")
    return false;
}
    if (num < 0 ){
        console.error(-1)
        return false
    }
    return true 
} 

function validArgumentLength(arguments) {
    if (arguments.length < 1){
        console.error("Ce script nécessite 1 argument.")
        return false
    }
    return true; 
}



//Utilitaire
function fibonacci(index) {
    let a = 0,  b = 1;
    for (let i = 2; i <= index; i++){
        let c = a + b;
        a = b;
        b = c;
    }
    return b; 
}

// Resolution 
function chekIfFibonacci(arguments){
    return fibonacci(arguments[0])
}


// Afficher les resultats 
function display() {
    const args = getArguments();
    if (!isValidArguments(args)){
        return;
    }
    if (!validArgumentLength){
        return;
    }
  
    const index = Number(args[0]);

    console.log(fibonacci(index));
}
display();